### PR TITLE
0box API tests base

### PR DESCRIPTION
**Pull request by stewartie4**
This PR extends the existing API test framework to support testing 0box.

Features added:
- Created a new 0box api client for abstraction
- Integrated with firebox
- Integrated with wallet create, list and delete APIs
- Integrated with CSRF API
- Combined the above for test teardown 
- Multiple bugs and inconsistencies have already been documented (search for FIXME) - I will raise issues for these 

Limitations:
- We only have a single test phone number meaning the environment needs to be reset every test case
- All 0box tests must run sequentially for the same reason 
- Firebase connectivity is required, we need to ensure we are not using up quotas 

  
Actions: https://github.com/0chain/actions/pull/95

### CODE REVIEW

File: internal/api/util/client/zbox_client.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+		Dst: &zboxWallets,
+		Headers: map[string]string{
+			"X-App-Client-ID":    "31f740fb12cf72464419a7e860591058a248b01e34b13cbf71d5a107b7bdc1e9",
+			"X-App-Client-Key":   "b6d86a895b9ab247b9d19280d142ffb68c3d89833db368d9a2ee9346fa378a05441635a5951d2f6a209c9ca63dc903353739bfa8ba79bad17690fe8e38622e96",
+			"X-App-Timestamp":    "1618213324",
+			"X-App-ID-TOKEN":     idToken,
+			"X-App-Phone-Number": phoneNumber,
+			"X-CSRF-TOKEN":       csrfToken,
+			"X-APP-TYPE":         "chimney",
+		}, //FIXME: List endpoint does not require signature!
```
Comment (Kishan-Dhakan) [inline internal/api/util/client/zbox_client.go:202]: Same as for above TODO

File: internal/api/util/test/system_test_framework.go [hunk 2]
```diff
@@ -90,7 +91,7 @@ func executeTest(s *SystemTest, name string, testFunction func(w *SystemTest), t
 
 func handlePanic(s *SystemTest) {
 	if err := recover(); err != nil {
-		s.Errorf("Test case exited due to panic - [%v]", err)
+		s.Errorf("Test case exited due to panic - [%v]", err, string(debug.Stack()))
```
Comment (Kishan-Dhakan) [inline internal/api/util/test/system_test_framework.go:94]: Needs to one more placeholder

File: tests/api_tests/0box_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	})
+
+	//FIXME: Missing field description does not match field name (Pascal case instead of snake case)
+	// [{ClientID  required } {PublicKey  required } {Timestamp  required } {TokenInput  required } {AppType  required } {PhoneNumber  required }]
+}
+
+func teardown(t *test.SystemTest, idToken, phoneNumber string) {
+	t.Logf("Tearing down existing test data for [%v]", phoneNumber)
+	csrfToken := createCsrfToken(t, zboxClient.DefaultPhoneNumber)
+	//wallets, _, _ := zboxClient.ListWallets(t, idToken, csrfToken, phoneNumber) //FIXME: list wallets endpoint returns the client id in place of the wallet id, and does not return the wallet id at all so we need to use list keys instead
```
Comment (Kishan-Dhakan) [inline tests/api_tests/0box_test.go:84]: Can remove this

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]
Amazing work, happy to merge post lint fixes

File: internal/api/model/zbox.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	LocalId      string `json:"localId"`
+	IsNewUser    bool   `json:"isNewUser"`
+	PhoneNumber  string `json:"phoneNumber"`
+}
+
+type CSRFToken struct {
+	CSRFToken string `json:"csrf_token"`
+}
+
+type ZboxWalletAlt struct { //TODO THIS IS INCONSISTENT AND SHOULD BE FIXED
```
[the remark this answers is not in the document]
Comment (stewartie4) [reply internal/api/model/zbox.go:19]: https://github.com/0chain/0box/issues/375

File: internal/api/util/client/zbox_client.go (hunk 1 above)
Comment (stewartie4) [reply internal/api/util/client/zbox_client.go:202]: https://github.com/0chain/0box/issues/376

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**